src/pybind/mgr/dashboard/frontend/cypress/integration/cluster/hosts.po.ts

```diff
@@ -28,4 +38,86 @@ export class HostsPageHelper extends PageHelper {
         expect(links_tested).gt(0);
       });
   }
+
+  @PageHelper.restrictTo(pages.index.url)
+  clickHostTab(hostname: string, tabName: string) {
+    this.getExpandCollapseElement(hostname).click();
+    cy.get('cd-host-details').within(() => {
+      this.getTab(tabName).click();
+    });
+  }
+
+  @PageHelper.restrictTo(pages.create.url)
+  add(hostname: string, exist?: boolean) {
+    cy.get(`${this.pages.create.id}`).within(() => {
+      cy.get('#hostname').type(hostname);
+      cy.get('cd-submit-button').click();
+    });
+    if (exist) {
+      cy.get('#hostname').should('have.class', 'ng-invalid');
+    } else {
+      // back to host list
+      cy.get(`${this.pages.index.id}`);
+    }
+  }
+
+  @PageHelper.restrictTo(pages.index.url)
+  checkExist(hostname: string, exist: boolean) {
+    this.getTableCell(this.columnIndex.hostname, hostname).should(($elements) => {
+      const hosts = $elements.map((_, el) => el.textContent).get();
+      if (exist) {
+        expect(hosts).to.include(hostname);
+      } else {
+        expect(hosts).to.not.include(hostname);
+      }
+    });
+  }
+
+  @PageHelper.restrictTo(pages.index.url)
+  delete(hostname: string) {
+    this.getTableCell(this.columnIndex.hostname, hostname).click();
+    this.clickActionButton('delete');
+    cy.get('cd-modal label[for="confirmation"]').click();
+    cy.get('cd-modal cd-submit-button').click();
+    cy.get(`${this.pages.index.id}`);
+  }
```

Comment:
Can't we just use the `delete` from `page-helper.po.ts`?

Reply:
I use `this.getTableCell(this.columnIndex.hostname, hostname).click();` to select a row only when a text in a specified column is matched. I've refactor `delete` in `page-helper.po.ts` a bit, please take a look. Thanks!
https://github.com/ceph/ceph/pull/36365/files#diff-458db60bec0f74509bd316ce0424f1dbR278